api/lib/infrastructure/repositories/knowledge-element-repository.js

```diff
@@ -90,19 +90,22 @@ module.exports = {
   },
 
   getSumOfPixFromUserKnowledgeElements(userId) {
-    return Bookshelf.knex.with('earnedPixWithRankPerSkill',
-      (qb) => {
-        qb.select('earnedPix', Bookshelf.knex.raw('ROW_NUMBER() OVER (PARTITION BY ?? ORDER BY ?? DESC) AS rank', ['skillId', 'createdAt']), 'competenceId')
-          .from('knowledge-elements')
-          .where({ userId });
-      })
-      .sum('earnedPix AS earnedPix')
-      .from('earnedPixWithRankPerSkill')
-      .where({ rank: 1 })
-      .groupBy('competenceId')
-      .then((pixEarnedByCompetence) => {
-        const pixScoreByCompetence = _.map(pixEarnedByCompetence, (pixEarnedForOneCompetence) =>  pixEarnedForOneCompetence.earnedPix);
-        return scoringService.totalUserPixScore(pixScoreByCompetence);
+    const earnedPixByCompetenceQuery = `
+      WITH earnedPixWithRankPerSkill as (
+        select "earnedPix", "competenceId", ROW_NUMBER() OVER (PARTITION BY "skillId" order by "createdAt" DESC) AS rank
+        from "knowledge-elements"
+        where "userId"=${userId}
+      )
+      SELECT "competenceId", sum("earnedPix") AS "earnedPix"
+      FROM earnedPixWithRankPerSkill
+      WHERE "rank" = 1
+      GROUP BY "competenceId"
+    `;
+    return Bookshelf.knex.raw(earnedPixByCompetenceQuery)
+      .then(({ rows }) => {
+        return _.sumBy(rows, function(pixEarnedForOneCompetence) {
+          return _.floor(scoringService.getBlockedPixScore(pixEarnedForOneCompetence.earnedPix));
+        });
```

Comment:
Du coup `totalUserPixScore` n'est plus utilisée ?
Je trouverais ça plus logique pour garder toute la logique de calcul au même endroit (`scoringService`), et donc de faire 
```suggestion
      .then(({ rows }) => scoringService.totalUserPixScore(rows));
```
avec 
```
function totalUserPixScore(pixEarnedByCompetences) {
        return _.sumBy(pixEarnedByCompetences, (pixEarnedForOneCompetence) => _.floor(scoringService.getBlockedPixScore(pixEarnedForOneCompetence.earnedPix));
```
dans le `scoringService`